test/sql/test_substrait_nested_types.test

```diff
@@ -0,0 +1,46 @@
+# name: test/sql/test_substrait_nested_types.test
+# description: Test nested type support in substrait extension (LIST, MAP, ARRAY)
+# group: [sql]
+
+require substrait
+
+statement ok
+PRAGMA enable_verification
+
+statement ok
+CREATE TABLE nested_test (
```

Comment:
I'd like to see additional nesting (a struct, list, or map within one of the others).